config.toml.example

```diff
@@ -1,40 +0,0 @@
-# sample config for interlock
```

Comment:
I moved it to `docs/examples/**/config.toml`

I felt it's better to have this example alongside all of the other examples so users only have to go to one place to find examples.

Also, having only one example in the root is a bit confusing because it has to cover multiple backends. For example, the nginx extension below includes a number of config options like `ConnectTimeout` and `ServerTimeout` that are specific to haproxy.